libr/bin/format/elf/elf.c

```diff
@@ -2076,15 +2258,26 @@ static RBinElfSymbol* get_symbols_from_phdr (struct Elf_(r_bin_elf_obj_t) *bin,
 	if (addr_sym_table > bin->size || addr_sym_table + size > bin->size)
 		goto beach;
 
+	for (i = 0; i < nsym; i++) {
+		r = r_buf_read_at (bin->b, addr_sym_table + i * sizeof (Elf_(Sym)),
+					s, sizeof (Elf_(Sym)));
+		if (r < 1) goto beach;
 #if R_BIN_ELF64
-	r = r_buf_fread_at (bin->b, addr_sym_table, (ut8*)sym,
-			bin->endian ? "I2cS2L": "i2cs2l", nsym);
+		sym[i].st_name = r_read_ble32 (s, bin->endian);
```

Comment:
I don't think you can increment i inside a braced initializer...

On 27 August 2016 11:22:40 AM AEST, radare <notifications@github.com> wrote:
>>  #if R_BIN_ELF64
>> -	r = r_buf_fread_at (bin->b, addr_sym_table, (ut8*)sym,
>> -			bin->endian ? "I2cS2L": "i2cs2l", nsym);
>> +		sym[i].st_name = r_read_ble32 (s, bin->endian);
>
>also.. what do you think about just:
>
>sym[i] = { .st_name = READ32(s), ...};
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/radare/radare2/pull/5592/files/780049f4b629688cbae35724c947fd5c41086067#r76507431

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.